### PR TITLE
Fix issue 3876, Fix checking enabled ray tracing features

### LINKED ISSUE #3876: False positive - VUID-vkGetAccelerationStructureBuildSizesKHR-rayTracingPipeline-03617
**Describe the Issue**
Getting a validation error when not enabling rayQuery features, even though rayTracingPipeline features are properly enabled. 
As per my understanding, this validation errors mentions needing either one of rayQuery or rayTracingPipeline features to be enabled, but it seems it really requires both (or only rayQuery). 

**Valid Usage ID**
VUID-vkGetAccelerationStructureBuildSizesKHR-rayTracingPipeline-03617

**Environment:**
 - OS: Linux 5.13.19-2-MANJARO
 - GPU: RTX 2080
 - SDK or header version if building from repo: v1.3.208
 - Options enabled (synchronization, best practices, etc.): VK_KHR_ray_tracing

**Additional context**

### PR DESCRIPTION
**Pull request by ziga-lunarg**
Closes #3876

### CODE REVIEW

File: tests/positive/other.cpp [hunk 1]
```diff
@@ -657,3 +657,57 @@ TEST_F(VkPositiveLayerTest, QueueThreading) {
 
     vk::QueueWaitIdle(queue_h);
 }
+
+TEST_F(VkPositiveLayerTest, ValidateGetAccelerationStructureBuildSizes) {
+    TEST_DESCRIPTION("Test enabled features for GetAccelerationStructureBuildSizes");
+
+    // Crashes without any warnings
+    if (IsDriver(VK_DRIVER_ID_AMD_PROPRIETARY)) {
```
Comment (ziga-lunarg) [inline tests/positive/other.cpp:665]: I wrote 2 different tests and they both crash inside AMD driver without any VUIDs being reported, I also checked the unimplemented ones.

Comment (jeremyg-lunarg) [reply tests/positive/other.cpp:670]: I wonder, does the AMD driver incorrectly require both features?

File: layers/parameter_validation_utils.cpp [hunk 2]
```diff
@@ -7895,9 +7895,8 @@ bool StatelessValidation::manual_PreCallValidateGetAccelerationStructureBuildSiz
     const auto *ray_tracing_pipeline_features =
         LvlFindInChain<VkPhysicalDeviceRayTracingPipelineFeaturesKHR>(device_createinfo_pnext);
     const auto *ray_query_features = LvlFindInChain<VkPhysicalDeviceRayQueryFeaturesKHR>(device_createinfo_pnext);
-    if (!(ray_tracing_pipeline_features || ray_query_features) ||
-        ((ray_tracing_pipeline_features && ray_tracing_pipeline_features->rayTracingPipeline == VK_FALSE) ||
-         (ray_query_features && ray_query_features->rayQuery == VK_FALSE))) {
+    if (!((ray_tracing_pipeline_features && ray_tracing_pipeline_features->rayTracingPipeline == VK_TRUE) ||
```
Comment (jeremyg-lunarg) [inline layers/parameter_validation_utils.cpp:7898]: I *think* this is correct but it still took me a minute.

**Review by jeremyg-lunarg — APPROVED** [review]
LGTM

File: tests/positive/other.cpp (hunk 1 above)
Comment (ziga-lunarg) [reply tests/positive/other.cpp:670]: Tried it, didn't help.

**Pull request merged**